gm4_soul_fragments/data/gm4_soul_fragments/functions/main.mcfunction

```diff
@@ -0,0 +1,19 @@
+# runs stuff on respawn
+execute as @e[type=player,scores={gm4_sf_rspwn=1..}] at @s run function gm4_soul_fragments:player/revived
+
+# manage players
+execute as @a[gamemode=!spectator] run function gm4_soul_fragments:player/submain
+
+# manage abandoned soul fragments
+execute as @e[type=minecraft:area_effect_cloud,tag=gm4_soul_fragment,tag=!gm4_new_soul_fragment] at @s run function gm4_soul_fragments:soul_fragment/process_check
```

Comment:
the `tag=!gm4_new_soul_fragment` isn't necessary because there should never be an AEC with that tag existing because it's removed the same tick that the AEC spawns